container/dangerzone.py

```diff
@@ -526,13 +370,25 @@ def main():
         print(f"Usage: {sys.argv[0]} [document-to-pixels]|[pixels-to-pdf]")
         return -1
 
-    converter = DangerzoneConverter()
+    job = ConversionJob()
 
     if sys.argv[1] == "document-to-pixels":
-        return converter.document_to_pixels()
+        try:
+            job.document_to_pixels()
+        except (RuntimeError, TimeoutError) as e:
+            job.update_progress(str(e), error=True)
+            return 1
+        else:
+            return 0  # Success!
 
-    if sys.argv[1] == "pixels-to-pdf":
-        return converter.pixels_to_pdf()
+    elif sys.argv[1] == "pixels-to-pdf":
+        try:
+            job.pixels_to_pdf()
+        except (RuntimeError, TimeoutError) as e:
+            job.update_progress(str(e), error=True)
+            return 1
+        else:
+            return 0  # Success!
```

Comment:
Maybe also the progress updating on error reporting should be done all within the converter object as it was before?

Reply:
That try/except format looks much cleaner, yeah! Not sure why GitHub is rendering your suggested change like that, though (might be confused by the `job.` renaming). It won't let me apply suggestions on deleted lines. Don't have time today to look as closely as I'd like, but welcome your thoughts.

`ValueError` should indeed be added to the tuple in `except (RuntimeError, TimeoutError) as e:` line.

Regarding the error reporting, all calls to `update_progress` within the `ConversionJob` class are for non-errors and all error reports are done in `main()`. This way, instead of doing this:

```
if something_is_wrong:
    update_progress("something went wrong", error=True)
    raise SomeError("something went wrong")
```

we can just write

```
if something_is_wrong:
    raise SomeError("something went wrong")
```

to get the same result. Either way, update_progress is always an instance method, so it's still specific to the job itself (so it won't cause any issues for batch processing).

Reply:
Out of paranoia, I suggest using `elif` instead of `if` on line 376. I'm not sure, off the top of my head, whether it would be possible, but we might worry that a very clever PDF that infects the first container (which runs the more dangerous `document_to_pixels` step) could sneakily change `sys.argv[1]`during runtime so that both steps end up running in the same container. Really, we shouldn't be allowing the first container to have access to the `/safezone` volume at all, but taking the extra precaution wouldn't hurt.

Reply:
> Out of paranoia, I suggest using elif instead of if on line 376.

> could sneakily change `sys.argv[1]` during runtime so that both steps end up running in the same container.

Well, I am working under the assumption that a malicious PDF can do whatever it wants in the container. It can run whatever routine it wants, including changing `sys.argv[1]`. But that won't change the fact that it's still running in the first container and there, it shouldn't have access to the `/safezone` volume (as per the container's run call: [this is the case](https://github.com/freedomofpress/dangerzone/blob/main/dangerzone/container.py#L82)).

But maybe I'm missing something in my assumptions.

`elif` if fine as well. Tbh I only put it because it because of text alignment OCD, but in reality `elif` would be more correct.

Reply:
Done in 625ac96.